### PR TITLE
NAS-121985 / 23.10 / Add Feedback Module

### PR DESCRIPTION
**Pull request by denysbutenko**
Work in progress. Appreciate suggestions on code improvements.

### CODE REVIEW

File: src/app/modules/ix-feedback/feedback-dialog/feedback-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    <ix-textarea
+      formControlName="message"
+      [label]="'Message' | translate"
+      [rows]="5"
+      [placeholder]="'Share your thoughts on our product\'s features, usability, or any suggestions for improvement.' | translate"
+    ></ix-textarea>
+  </form>
+
+  <div class="file-ticket-helper">
```
Comment (undsoft) [inline src/app/modules/ix-feedback/feedback-dialog/feedback-dialog.component.html:27]: Let's move this to the top of the dialog. Also `File Ticket` doesn't look good as button, let's make it a link.

File: src/app/modules/ix-feedback/ix-feedback.service.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+export class IxFeedbackService {
+  private readonly hostname = 'https://feedback.ui.truenas.com';
+  private readonly token = '946bde8d85fc86c6c48344adf1ef2139';
+  private headers = new HttpHeaders({
+    Authorization: `Bearer ${this.token}`,
+  });
+
+  constructor(private httpClient: HttpClient) {}
+
+  getReviews(): Observable<Review[]> {
```
Comment (undsoft) [inline src/app/modules/ix-feedback/ix-feedback.service.ts:19]: Let's not add methods we won't use

**Review by undsoft — APPROVED** [review]
Looks pretty good, nice work. 👍🏻

File: src/app/modules/ix-feedback/feedback-dialog/feedback-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  styleUrls: ['./feedback-dialog.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FeedbackDialogComponent implements OnInit {
+  protected form = this.formBuilder.group({
+    rating: [0, [Validators.required, rangeValidator(1, maxRatingValue)]],
+    message: [''],
+    release: [''],
+    environment: [FeedbackEnvironment.Development],
+    host_u_id: [UUID.UUID()],
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/modules/ix-feedback/feedback-dialog/feedback-dialog.component.ts:40]: Ok, let's update it later.

File: src/app/modules/layout/components/topbar/topbar.component.html [hunk 4]
```diff
@@ -38,6 +38,19 @@
...
+        *appIfNightly
+        mat-icon-button
+        matBadge="+"
+        matBadgeSize="small"
+        class="topbar-button-right"
+        ixTest="leave-feedback"
+        [matTooltip]="'How would you rate this page?' | translate"
+        (click)="onFeedbackIndicatorPressed()"
+      >
+        <ix-icon name="mdi-star-half-full"></ix-icon>
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/modules/layout/components/topbar/topbar.component.html]: image

**Pull request merged**